### PR TITLE
Fix for the eventual dialog freeze

### PR DESCRIPTION
**Pull request by skarbat**
* Trying to fix LightDM multiple instance freeze by creating
   one single instance and reconnecting events dynamically to each view

 * This fix is not complete, as it eventually freezes the message box
   after a failed login attempt, even if no multiple views have been opened
   (i.e. staying at the login view alone)

### CODE REVIEW

**Comment by alex5imon** [comment]
@skarbat what do you want to do with this?

**Comment by skarbat** [comment]
This is not resolving the problem and still freezes the greeter, I am not keen to merge it. Unfortunately this is an obscure problem after sharing with @tombettany I cannot find a solution for right now. @alex5imon

**Comment by skarbat** [comment]
The latest 2 changesets fix the freeze problem, meaning the user can switch between login & create new user account views without any problems until the login proceeds. However it is no 100% safe and it is possible to eventually freeze the LightDM Greeter.

cc @alex5imon @tombettany

**Comment by alex5imon** [comment]
Please merge so it can be tested tomorrow :+1:

File: setup.py [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-      url='https://github.com/KanoComputing/kano-greeter',
-      packages=['kano_greeter'],
-      scripts=['bin/kano-greeter', 'bin/kano-greeter-account'],
-      package_data={'kano_greeter': ['media/css/*']},
-      data_files=[
-          ('/usr/share/kano-greeter/wallpapers', setuptools.findall('wallpapers')),
-          ('/usr/share/xgreeters', ['config/kano-greeter.desktop']),
-          ('/var/lib/lightdm', ['config/.kdeskrc'])
-      ] + get_locales()
-     )
```
Comment (tombettany) [inline setup.py]: @skarbat Should `setup.py` have been removed? I think we use it for package creation.

cc @pazdera

Comment (skarbat) [inline setup.py]: my bad, thanks @tombettany ! Added it back.

**Pull request merged**